File: frontend/src/services/analytics.ts
import { get } from "./api";

export interface PostsPerCategory {
  kategori_id: number;
  kategori: string;
  antal_poster: number;
}

export interface PostsPerUser {
  anvandar_id: number;
  anvandarnamn: string;
  antal_poster: number;
}

export interface MostUsedConcept {
  begrepp_id: number;
  ord: string;
  antal_kopplingar: number;
}

export function fetchPostsPerCategory(): Promise<PostsPerCategory[]> {
  return get<PostsPerCategory[]>("/analytics/posts-per-category");
}

export function fetchPostsPerUser(): Promise<PostsPerUser[]> {
  return get<PostsPerUser[]>("/analytics/posts-per-user");
}

export function fetchMostUsedConcepts(): Promise<MostUsedConcept[]> {
  return get<MostUsedConcept[]>("/analytics/most-used-concepts");
}

File: backend/app/routers/analytics.py
"""Endpoints för analys."""
from fastapi import APIRouter

from app.repositories import analytics_repo

router = APIRouter()


@router.get("/analytics/posts-per-category")
def posts_per_category():
    """Antal poster per kategori."""
    return analytics_repo.get_posts_per_category()


@router.get("/analytics/posts-per-user")
def posts_per_user():
    """Antal poster per användare."""
    return analytics_repo.get_posts_per_user()


@router.get("/analytics/most-used-concepts")
def most_used_concepts():
    """Mest använda begrepp."""
    return analytics_repo.get_most_used_concepts()
